### DISCUSSION TITLE
Welcome to DeDRM_tools Discussions!

### DISCUSSION THREAD

**Discussion by apprenticeharper** [apprenticeharper/DeDRM_tools#1438] [General]
## 👋 Welcome!

General users should use discussions for questions about how to use the tools.

Contributors can use discussions to comment on outstanding bugs and arrange work sharing

Issues should now only be used for what appear to be genuine bugs.

Let's see how it goes...

Harper.

**Comment by toriannh** [comment]
Hello,

I want to thank you for creating this tool. 

Unfortunately I am not having great success with removing the DRM from the ebook file. I use Kindle for Mac and I am running on Mac Big Sur version 11.2 Beta. I am not sure what the problem is. I tried to DRM the file under the 6.8.1 and 6.8.0 releases with Calibre version 4.23 and it is not working. I would load the plugin from the file and the restart calibre, then when I add the file to calibre it the ebook still has the DRM to it. I tried to DRM the file under the 7.0 DRM update with the 5.0 version of Calibre and it is still not working for me. 

If you could kindly help me out here, that would be much appreciated. Thank you.

**Reply by ElleKayEm** [reply]
Stick with calibre 4.23 and DeDRM 6.8.1.  Your best bet is Method 2 at this post:  
https://www.mobileread.com/forums/showthread.php?t=283371

And also see the Kindle for Mac notes in this post:  
https://www.mobileread.com/forums/showpost.php?p=3819708&postcount=508

**Reply by toriannh** [reply]
Thank you. It worked!

I used the following command in the terminal: sudo chmod -x /Applications/Kindle.app/Contents/MacOS/renderer-test

I was using method 4 earlier but I see that did not work well for me.

**Comment by sarakpandy** [comment]
Hey - I am trying to deDRM a etextbook that I have purchased that is intended for Adobe Digital Editions.  I have downloaded Calibre version 5.8.1 and have tried using the new 7.0.3 plugin following the instructions found in these forums.  I am working on a macbook, but have access to a windows computer if what I am trying to do is not possible on a mac.  

I have downloaded the zip plugin file, and opened in through the Preferences > Change Calibre Behaviour > Advanced > Plugins > Load Plugin from the file where it lets it be select and open the zipped version of the plugin.  From here on out, I accept all of the things, then click apply and it appears in my list of Files as a nice green jigsaw puzzle.  I have then restarted the application, as well as my computer once throughout my multiple times trying.  

This next step I am trying to add my book to Calibre.  I click Add Book and I go and find the file INSIDE the documents folder (Which apple stores on the cloud), under the Adobe Digital Editions. It is here that I seem to have something different show up then other people on youtube.  I have a pdf version of the ebook in that folder (no ePub - I had previously downloaded the .acsm file from the publisher and it seems that Digital editions rendered it and saved it as that pdf.  

Then when I add the ebook through this method, it shows up in calibre but I still cannot open it.  Does anyone have any insight into what I could be doing wrong? I would appreciate any help I can get - I am new at this! Thanks!

**Reply by ElleKayEm** [reply]
ADE is on the same machine and user as calibre with DeDRM?  ADE was authorized with your Adobe ID before you opened the acsm file?  If all that's the case, it may just be that the beta isn't working for your particular file yet.  It should work with calibre 4.23 and DeDRM 6.8.1.

**Comment by vimy100** [comment]
Greetings.  I had a DeDRM issue a while back (I'm not all that computer savvy) and the folks here (ElleKayEm actually) were very helpful and now, 4 years later, another problem has cropped up and I am hoping you can help me once again.  I just noticed this morning (23 Feb 2021) that DRM has failed on the last few books I tried to convert in Calibre, my last successful conversion occurring 8 days ago (15 Feb 2021).  So far as I can tell nothing has changed in regard to my apps and hardware.

I use Windows 10 Pro (still with ver 1809), Calibre 4.23, Kindle for PC 1.17.1, and DeDRM_tools_6.8.0.   

The error message from Calibre states:
File "site-packages\calibre\ebooks\mobi\reader\mobi6.py", line 158, in check_for_drm
DRMError: 

Do you have any thoughts on why this would suddenly happen?

**Reply by ElleKayEm** [reply]
Kindle for PC has been unexpectedly updating on people lately.

**Reply by vimy100** [reply]
Yes, I checked and my Kindle for PC was last updated in Oct 2020 and I have successfully converted many books since then and the version is still 1.17.1.  
However, I just noticed that the books which failed to convert were all AZW format.  Most of the books that I have converted in recent years have been either AZW3 or AZW4.  I haven't seen plain old AZW for a long time and don't know why these books, all recent purchases, should not be AZW3/4.  Might that be an issue?

**Reply by ElleKayEm** [reply]
AZW is the generic extension Kindle for PC uses for all Amazon formats.  Only when DeDRM has successfully removed the DRM do you see the actual format extension in calibre.  So seeing AZW in calibre just means DeDRM didn't work.  It could be AZW3 or something else.  Let's take a look at a log:  https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

**Reply by vimy100** [reply]
Now this is odd.  I followed the instructions at the link and the log indicated that the DeDRM procedure was successful and sure enough the format was now AZW3.  I tried a conversion and it worked fine.  Here s the log file.
calibre Debug log
calibre 4.23 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.17763-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.17763')
Python 2.7.16+
Windows: ('10', '10.0.17763', 'SP0', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (6, 8, 0) && Obok DeDRM (6, 7, 0)
calibre 4.23 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.17763-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.17763')
Python 2.7.16+
Windows: ('10', '10.0.17763', 'SP0', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (6, 8, 0) && Obok DeDRM (6, 7, 0)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 127.664921466 x 127.590697674
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[1.98] splash screen shown
[1.98] Initializing db...
[5.03] db initialized
[5.03] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.3 obok::action_err.py - loading translations
[13.66] main UI initialized...
[13.66] Hiding splash screen
[69.45] splash screen hidden
[69.45] Started up in 69.45 seconds with 9243 books
DeDRM v6.8.0: Trying to decrypt Hope for the Brokenhearted_ God - John Luke Terveen.azw
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Hope for the Brokenhearted: God's Voice of Comfort in the Midst of Grief and Loss
Found 4 keys to try after 0.9 seconds
Crypto Type is: 2
File is encoded with PID nHchflmv4Q.
Decrypting. Please wait . . . . done
Decryption succeeded after 0.9 seconds
DeDRM v6.8.0: Finished after 0.9 seconds
Added Hope for the Brokenhearted: God's Voice of Comfort in the Midst of Grief and Loss to db in: 2.4
Added 1 books in 6.1 seconds

I then thought that perhaps something was fixed and so tried the others that had failed but they still failed.  Here's an example:

Convert book 4 of 4 (The Cross Before Me: Reimagining the Way to the...)
Conversion options changed from defaults:
  pdf_serif_family: u'MS Shell Dlg 2'
  cover: u'C:\\Users\\blmj\\AppData\\Local\\Temp\\calibre_lqoitn\\_km_jx.jpeg'
  verbose: 2
  pdf_sans_family: u'MS Shell Dlg 2'
  read_metadata_from_opf: u'C:\\Users\\blmj\\AppData\\Local\\Temp\\calibre_lqoitn\\pkiplp.opf'
  output_profile: u'generic_eink'
Resolved conversion options
calibre version: 4.23.0
{'asciiize': False,
 'author_sort': None,
 'authors': None,
 'base_font_size': 0.0,
 'book_producer': None,
 'change_justification': u'original',
 'chapter': u"//*[((name()='h1' or name()='h2') and re:test(., '\\s*((chapter|book|section|part)\\s+)|((prolog|prologue|epilogue)(\\s+|$))', 'i')) or @class = 'chapter']",
 'chapter_mark': u'pagebreak',
 'comments': None,
 'cover': u'C:\\Users\\blmj\\AppData\\Local\\Temp\\calibre_lqoitn\\_km_jx.jpeg',
 'custom_size': None,
 'debug_pipeline': None,
 'dehyphenate': True,
 'delete_blank_paragraphs': True,
 'disable_font_rescaling': False,
 'duplicate_links_in_toc': False,
 'embed_all_fonts': False,
 'embed_font_family': None,
 'enable_heuristics': False,
 'expand_css': False,
 'extra_css': None,
 'filter_css': u'',
 'fix_indents': True,
 'font_size_mapping': None,
 'format_scene_breaks': True,
 'html_unwrap_factor': 0.4,
 'input_encoding': None,
 'input_profile': <calibre.customize.profiles.InputProfile object at 0x000002D2399ADE88>,
 'insert_blank_line': False,
 'insert_blank_line_size': 0.5,
 'insert_metadata': False,
 'isbn': None,
 'italicize_common_cases': True,
 'keep_ligatures': False,
 'language': None,
 'level1_toc': None,
 'level2_toc': None,
 'level3_toc': None,
 'line_height': 0.0,
 'linearize_tables': False,
 'margin_bottom': 5.0,
 'margin_left': 5.0,
 'margin_right': 5.0,
 'margin_top': 5.0,
 'markup_chapter_headings': True,
 'max_toc_links': 50,
 'minimum_line_height': 120.0,
 'no_chapters_in_toc': False,
 'no_inline_navbars': False,
 'output_profile': <calibre.customize.profiles.GenericEink object at 0x000002D2399BDAC8>,
 'page_breaks_before': u"//*[name()='h1' or name()='h2']",
 'paper_size': u'letter',
 'pdf_add_toc': False,
 'pdf_default_font_size': 20,
 'pdf_footer_template': None,
 'pdf_header_template': None,
 'pdf_hyphenate': False,
 'pdf_mark_links': False,
 'pdf_mono_family': u'Courier',
 'pdf_mono_font_size': 16,
 'pdf_odd_even_offset': 0.0,
 'pdf_page_margin_bottom': 72.0,
 'pdf_page_margin_left': 72.0,
 'pdf_page_margin_right': 72.0,
 'pdf_page_margin_top': 72.0,
 'pdf_page_number_map': None,
 'pdf_page_numbers': False,
 'pdf_sans_family': u'MS Shell Dlg 2',
 'pdf_serif_family': u'MS Shell Dlg 2',
 'pdf_standard_font': u'serif',
 'pdf_use_document_margins': False,
 'prefer_metadata_cover': False,
 'preserve_cover_aspect_ratio': False,
 'pretty_print': False,
 'pubdate': None,
 'publisher': None,
 'rating': None,
 'read_metadata_from_opf': u'C:\\Users\\blmj\\AppData\\Local\\Temp\\calibre_lqoitn\\pkiplp.opf',
 'remove_fake_margins': True,
 'remove_first_image': False,
 'remove_paragraph_spacing': False,
 'remove_paragraph_spacing_indent_size': 1.5,
 'renumber_headings': True,
 'replace_scene_breaks': u'',
 'search_replace': '[]',
 'series': None,
 'series_index': None,
 'smarten_punctuation': False,
 'sr1_replace': None,
 'sr1_search': None,
 'sr2_replace': None,
 'sr2_search': None,
 'sr3_replace': None,
 'sr3_search': None,
 'start_reading_at': None,
 'subset_embedded_fonts': False,
 'tags': None,
 'timestamp': None,
 'title': None,
 'title_sort': None,
 'toc_filter': None,
 'toc_threshold': 6,
 'toc_title': None,
 'transform_css_rules': '[]',
 'uncompressed_pdf': False,
 'unit': u'inch',
 'unsmarten_punctuation': False,
 'unwrap_lines': True,
 'use_auto_toc': False,
 'use_profile_size': False,
 'verbose': 2}
InputFormatPlugin: MOBI Input running
on C:\Users\blmj\AppData\Local\Temp\calibre_lqoitn\ysyl6d.azw
Found KF8 MOBI of type u'standalone'
Python function terminated unexpectedly
  The Cross Before Me: Reimagining the Way to the Good Life (Error Code: 1)
 Traceback (most recent call last):
  File "site.py", line 114, in main
  File "site.py", line 88, in run_entry_point
  File "site-packages\calibre\utils\ipc\worker.py", line 209, in main
  File "site-packages\calibre\gui2\convert\gui_conversion.py", line 43, in gui_convert_override
  File "site-packages\calibre\gui2\convert\gui_conversion.py", line 28, in gui_convert
  File "site-packages\calibre\ebooks\conversion\plumber.py", line 1110, in run
  File "site-packages\calibre\customize\conversion.py", line 246, in __call__
  File "site-packages\calibre\ebooks\conversion\plugins\mobi_input.py", line 47, in convert
  File "site-packages\calibre\ebooks\mobi\reader\mobi8.py", line 86, in __call__
  File "site-packages\calibre\ebooks\mobi\reader\mobi6.py", line 158, in check_for_drm
DRMError: The Cross Before Me: Reimagining the Way to the Good Life

So when the DeDRM failed I then tried the debug procedure with this second book and once again everything worked as it is supposed to work.  So it appears that I can successfully DeDRM a Kindle book and convert it to pdf only by running the debug feature on Calibre.  What do you think about that?  :-)

**Comment by ElleKayEm** [comment]
So if you add a book to calibre without it being in debug mode, it doesn't get decrypted -- the extension stays AZW?

**Reply by vimy100** [reply]
Yes, that is right.

**Reply by ElleKayEm** [reply]
You could try updating to calibre 5.x and DeDRM 7.1.0.

**Reply by vimy100** [reply]
Will do.  I'll let you know how it transpires.

**Reply by vimy100** [reply]
@ ElleKayEm  I have implemented your suggestion to try Calibre 5.x and DeDRM 7.1.0 and the conversion worked without fail in every case I tried (four books).  I think I'm back on track :-).
I am sure you and Apprentice Alf and apprenticeharper have been thanked many times for the tremendous work you have done but I would like to thank you once again.  You have been an immense help to me more than once and the long hours you three (and perhaps many others as well) have devoted free of charge to this project is quite remarkable.  I also appreciate so much the gentleness and patience with which you interact with folks who do not have much by way of computer skills. The helpful and intelligent way in which you have organized the tutorial files is also a blessing.  You bunch are wonderful.  Thank you from the bottom of my heart for all the help you have been to so many of us.

**Comment by brocaleb** [comment]
Can't remove DRM from mooink / readmoo

**Reply by ElleKayEm** [reply]
I'll just repeat what I said over in the issue you opened:

These tools don't support that right now, but it looks like there are some.  Here's one: https://github.com/dumo-dedrm/readmoo-dedrm

A quick search here at GitHub for 'readmoo' turns up others.  No experience with them.  This the first I've head of it.

**Comment by michaelbr51** [comment]
Hi:
When I was using Windows 8.1, I did DeDRM without problem. Now I'm using MX Linux, tried again to remove DRM, it did not work, here's my environment
- OS MX Linux 19.4
- Calibre 5.16.1
- DeDRM plugin 7.2.1
According to what I read over the net, when added a DRM ebook, the DRM will be removed automatically. After added the DRM epub file in Calibre (did not give any error), and tried to open with Calibre's ebook viewr, it said "This book is locked by DRM", am I missing some steps? What else I need to do to solve this error?
![Calibre_DeDRM](https://user-images.githubusercontent.com/20541206/116813339-9f689a80-ab53-11eb-92b9-283d74ed2231.png)

**Reply by ElleKayEm** [reply]
If you're decrypting books from Kindle for PC or Adobe Digital Editions, DeDRM needs to be able to grab the key.  On Windows and Mac, this happens when the programs are installed on the same machine and user.  With Linux, this can be tricky.  One workaround is to temporarily install Windows calibre with DeDRM where you have the programs installed, decrypt a book there, save the key, and finally import that key into your Linux calibre/DeDRM's customization.

**Comment by cygnusdave** [comment]
hi - read a bunch of threads but still can't get amazon book to convert in calibre anymore

I'm running Kindle Version 1.17.1 (44183) for PC. Calibre 5.10.1. I've downloaded the DRM Plugins (and redownloaded the newest versions). Everything works fine when importing books from Adobe Digital Editions.

When downloading from Kindle I get cannot convert due to DRM message. I've tried to directly download the Kindle book through download manually and send through USB as well as by opening book up in the kindle app for PC and then finding the file.

I did the import while debugging as recommended and the file is attached

any insight in most appreciated

[tmp0q07g9v1.txt](https://github.com/apprenticeharper/DeDRM_tools/files/6417106/tmp0q07g9v1.txt)

**Reply by ElleKayEm** [reply]
Kindle for PC/Mac used to download the book needs to be installed on the same machine and user as calibre with DeDRM.  So in your log, I see a book from the Kindle app, but DeDRM is not finding any installation of Kindle for Mac to retrieve the key from.

If you download a book from the website, you need to choose an E-ink Kindle (not a Fire) for the device and have the Kindle's serial number entered into DeDRM's customization.

**Comment by zoiando** [comment]
hello thank you so much for the tool, it's working perfectly. I have an ebook from ridi-books and i can't convert it, could you please add this kind of epub in the tools or say to me how can i convert with other plugin?

**Reply by ElleKayEm** [reply]
Not aware of any tool which removes that DRM.

**Comment by huguiyo** [comment]
Hi, I'm sorry.. certainly you told a million times but... I cant dedrm a kindle ebook. "Is locked by DRM" says... is my problem that calibre (and new plugins) are installed in a device that is not that the owner of the ebook? Im sorry for my english... Thanks..

**Reply by ElleKayEm** [reply]
If the book was downloaded with Kindle for PC/Mac, then calibre with DeDRM needs to be on the same computer and user.

**Comment by Ninko85** [comment]
Hi,
I'm very new to trying to remove DRM from my Kindle books and I'm not having much luck.
When I add books to Calibre there is no information/cover for it, and when I convert it and try to open it, it's all garbled.

I'm using
Calibre 5.24
Apprentice Alf’s DeDRM 7.2.1
Kindle for PC 1.32.0

I'm probably doing something wrong, can anyone help me please?
Thanks

**Reply by ElleKayEm** [reply]
See this post:  https://www.mobileread.com/forums/showthread.php?t=283371

**Comment by ajbigio** [comment]
There is a book that when opening provides this as the last comment on why it cannot open the book:
ImportError: The _imaging extension was built for another version of Pillow or PIL:
Core version: 8.3.1
Pillow version: 7.2.0
I can add details If necessary. There is only one book that has presented this issue.
Is there any workaround?
Thanks for everything that you do.

**Reply by ElleKayEm** [reply]
Let's see a log of adding the book.
